collect_app/src/main/java/org/odk/collect/android/activities/FormChooserList.java

```diff
@@ -44,13 +46,13 @@
  * @author Yaw Anokwa (yanokwa@gmail.com)
  * @author Carl Hartung (carlhartung@gmail.com)
  */
-public class FormChooserList extends FormListActivity implements DiskSyncListener, AdapterView.OnItemClickListener {
+public class FormChooserList extends FormListActivity implements
+        DiskSyncListener, AdapterView.OnItemClickListener, LoaderManager.LoaderCallbacks<Cursor> {
```

Comment:
Can we make this change without adding multiple LoaderCallbacks implementations, consisting of mostly duplicate code? This seems to be adding a lot of technical debt.

Reply:
I spent some time trying to factor out common code into something like this:

![screen shot 2018-03-24 at 6 44 13 pm](https://user-images.githubusercontent.com/55099/37870814-9fcf84aa-2f93-11e8-948e-3484fbd8e0a0.png)

![screen shot 2018-03-24 at 6 45 20 pm](https://user-images.githubusercontent.com/55099/37870818-a72151d4-2f93-11e8-9276-2ae524529211.png)

But the subclasses would have to do enough overriding as to make this probably not worth the trouble. 

It would be good to keep a goal of some sort of unification of these classes in mind as we go forward.

Reply:
I put an artificial delay in one of the background methods and was able to see the progress bar. It’s pretty.

Reply:
You can even use this to load 1000+ instances to see the progress bar
https://github.com/opendatakit/collect/files/1155963/backup1249.zip

Reply:
> Can we make this change without adding multiple LoaderCallbacks implementations, containing some duplicate code?

My intention was to move all the duplicate code to AppListActivity but there are two activities who do not use cursor loaders. But I really like the idea of removing the duplicate code but it is only able to remove the duplicate `onLoaderReset` while rest is the same. Is there a way to make it more compact?